assets/src/infrastructure/ui/display/DebugInfo.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback } from 'react';
+import { Button, Collapse } from '@chakra-ui/core';
+import styled from '@emotion/styled';
+
+const Pre = styled.pre`
+	border-radius: 5px;
+	box-sizing: border-box;
+	padding: 1em 2em;
+	color: #a9ce47;
+	background-color: #26203d;
+`;
+
+interface DebugInfoProps {
+	data: any;
+	asJson?: boolean;
+	asCollapse?: boolean;
+}
+
+const DebugInfo: React.FC<DebugInfoProps> = ({ data, asJson = true, asCollapse = true }) => {
+	const [show, setShow] = React.useState(false);
+	const isDev = !process.env.NODE_ENV || process.env.NODE_ENV === 'development';
+
+	if (!isDev) {
+		return null;
+	}
+
+	const dataToRender = asJson ? JSON.stringify(data, null, 2) : data;
+
+	const output = <Pre>{dataToRender}</Pre>;
+
+	if (!asCollapse) {
+		return output;
+	}
+
+	const handleToggle = useCallback(() => setShow(!show), [show]);
+
+	return (
+		<>
+			<Button size='sm' onClick={handleToggle} mt='1rem'>
+				{show ? 'Hide' : 'Show'} Debug Info
+			</Button>
+			<Collapse mt={4} isOpen={show}>
+				{output}
+			</Collapse>
+		</>
```

Comment:
too specific for an adapter.

turn this into something like:

- Collapsable

- ShowHideContainer

- ???

and **THEN** have the `DebugInfo` component in the app layer, import this and use it for doing its thing.

The way you have this right now, this logic can **only** be used for displaying debug data, whereas this adapter should be usable anywhere we want to have some content that can be hidden and displayed via a toggle, such as the filter bar filters. As is, you would what? Create another component in the infrastructure layer for displaying the filter bar filters that uses the same Chakra components? 

The things in this layer need to be completely generic but completely configurable so that they can be utilized in any other layer

Reply:
updated, but let's not abstract away the logic into `ShowHideContainer`, because:
1. each component standalone would be useless
2. I'd still have to `show` prop around, this is too much of abstraction IMHO

Reply:
your pong on this?

Reply:
oh no I didn't mean to split the button out from the `Collapse` container...

I just meant to make this generic so that any use case that requires a container that can be toggled in and out via a button could then use this component, whereas now it's defined as `DebugInfo` which limits the uses (even if just in name)

Reply:
ok, these were separated, lemme know if that's fine